cmd/deploy/command_configuration.go

```diff
@@ -162,17 +171,16 @@ func switchSSHRepoToHTTPS(repo string) (*url.URL, error) {
 	if err != nil {
 		return nil, err
 	}
-	if repoURL.Scheme == "ssh" {
-		repoURL.Scheme = "https"
+	switch repoURL.Scheme {
+	case sshScheme, httpScheme:
+		repoURL.Scheme = httpsScheme
 		repoURL.User = nil
-		repoURL.Path = strings.TrimSuffix(repoURL.Path, ".git")
```

Comment:
in the case of `ssh` there was a trim of `.git` which i removed in the refactor, i think this was not needed because the repository url is transformed to https which will require the `.git` isnt it?

Reply:
I don't think https requires the .git suffix. Both ways (with and without) should work the same. I think in some places in the code we are removing it to check if 2 repos are the same or not

Reply:
found https://github.com/okteto/okteto/blob/1a94bc9816077ee0aff8f8554e8c640380099b68/pkg/okteto/pipeline.go#L231 where a comparison is made, but is not related with the func refactored here.

Rarding the `/` thing mentioned at the linked func, in this case there is also no need to remove, as when the url is transformed to string this path will be correctly parsed.